# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/TestingFunctions.cpp
# Commit: 1a3bc285395f
# Full Hash: 1a3bc285395ffcc5d9281dd45d4f6c2bb918bf43
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-03-04 04:58:00
# Regressor Bug: 1861078
# File Overlap Count: 1
# Description:
#   Bug 1861078 - wasm: make metadata offsets relative to CodeBlock->codeBase. r=rhunt
#   
#   Metadata used offsets to the beginning of code segment. Change that to use
#   offsets relative to the beginning of CodeBlock. This allows us to avoid offseting
#   metadata later on and prepare the work for simplification of stackmaps.
# ==============================================================================

diff -r 66d70b730ea6 -r 1a3bc285395f js/src/builtin/TestingFunctions.cpp
--- a/js/src/builtin/TestingFunctions.cpp	Mon Mar 03 16:15:17 2025 +0000
+++ b/js/src/builtin/TestingFunctions.cpp	Mon Mar 03 16:37:47 2025 +0000
@@ -1780,12 +1780,11 @@
     js::wasm::Instance& inst = fun->wasmInstance();
     const uint32_t funcIndex = fun->wasmFuncIndex();
     const js::wasm::CodeBlock& codeBlock = inst.code().funcCodeBlock(funcIndex);
-    const js::wasm::CodeSegment& segment = *codeBlock.segment;
     const js::wasm::FuncExport& func = codeBlock.lookupFuncExport(funcIndex);
     const js::wasm::CodeRange& codeRange = codeBlock.codeRange(func);
 
-    jit_begin = segment.base() + codeRange.begin();
-    jit_end = segment.base() + codeRange.end();
+    jit_begin = codeBlock.base() + codeRange.begin();
+    jit_end = codeBlock.base() + codeRange.end();
   } else if (fun->hasJitScript() && fun->nonLazyScript()->hasIonScript()) {
     sprinter.printf("; backend=ion\n");
     jit_begin = fun->nonLazyScript()->ionScript()->method()->raw();